psycopg3/transaction.py

```diff
@@ -50,29 +88,48 @@ def __exit__(
         exc_tb: Optional[TracebackType],
     ) -> None:
         with self._conn.lock:
-            if exc_type is None:
+            if exc_type is None and not self.force_rollback:
+                # Commit changes made in the transaction context
                 if self._savepoint_name:
                     self._conn._savepoints.pop()
                     # TODO: Add test for this assert
                     # assert self._conn._savepoints.pop() == self._savepoint
-                    self._conn._exec_command(
-                        b"release savepoint " + self._savepoint_name
+                    self._log_and_exec_command(
+                        b"RELEASE SAVEPOINT " + self._savepoint_name
                     )
                 if self._outer_transaction:
                     # TODO: Add test for this assert
                     # assert len(self._conn._savepoints) == 0
-                    self._conn._exec_command(b"commit")
+                    self._log_and_exec_command(b"COMMIT")
                     self._conn._savepoints = None
             else:
+                # Rollback changes made in the transaction context
+                if exc_type is Rollback:
```

Comment:
`issubclass(exc_type, Rollback)` would be better here, which is similar to what `except Rollback` would do.

Reply:
I'll use `if isinstance(exc_val, Rollback):` to match what I'm doing below to satisfy mypy (since it doesn't know the relationship between `exc_type` and `exc_val`).